src/base/Group.h

```diff
@@ -10,10 +10,19 @@ namespace Groups
 {
 
 template <typename T>
-class Group : private std::set<T*>
+struct IdCompare
+{
```

Comment:
Can we think of a more expressive name? 

I would think that if `IdCompare(a,b)` yields `true`, it implies that `a==b` which is not correct.